src/object.js

```diff
@@ -146,11 +160,43 @@ ReflectionObject.prototype.onRemove = function onRemove(parent) {
 ReflectionObject.prototype.resolve = function resolve() {
     if (this.resolved)
         return this;
-    if (this.root instanceof Root)
-        this.resolved = true; // only if part of a root
+    if (this.root instanceof Root || this.parent) {
```

Comment:
I think we also need to make sure that parent.resolved is true.  If it's not, we can't resolve features yet